atmo/jobs/jobs.py

```diff
@@ -1,30 +1,65 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, you can obtain one at http://mozilla.org/MPL/2.0/.
+from django.db import transaction
+import logging
 import newrelic.agent
 
 from .models import SparkJob
+from atmo.clusters.provisioners import ClusterProvisioner
+
+logger = logging.getLogger(__name__)
 
 
 @newrelic.agent.background_task(group='RQ')
 def run_jobs():
     """
     Run all the scheduled tasks that are supposed to run.
     """
+    # first let's update the job statuses if there are prior runs
     run_jobs = []
-    for job in SparkJob.objects.all():
-        # first let's update the status if there is a prior run
-        if job.latest_run:
-            job.latest_run.update_status()
-
-        # then let's check if the job should be run at all
-        if job.should_run():
-            job.run()
-            run_jobs.append(job.identifier)
-
-        # and then check if the job is expired and terminate it if needed
-        if job.is_expired:
-            # This shouldn't be required as we set a timeout in the bootstrap script,
-            # but let's keep it as a guard.
-            job.terminate()
+    jobs = SparkJob.objects.all()
+
+    with transaction.atomic():
+        # get the jobs with prior runs
+        jobs_with_runs = jobs.filter(runs__isnull=False).prefetch_related('runs')
```

Comment:
so here we are fetching all the jobs with at least a run, no matter the status of the run, right? Shouldn't we fetch only those with at least a run in a non final status?